index.js

```diff
@@ -27,12 +29,22 @@ function NYC (opts) {
   this._istanbul = opts.istanbul
   this.subprocessBin = opts.subprocessBin || path.resolve(__dirname, './bin/nyc.js')
   this._tempDirectory = opts.tempDirectory || './.nyc_output'
-  this.cwd = opts.cwd || process.env.NYC_CWD || process.cwd()
-  this.reporter = arrify(opts.reporter || 'text')
+
+  var cwd = opts.cwd || process.env.NYC_CWD || process.cwd()
+  var pkgPath = pkgUp.sync(cwd)
 
   // you can specify config in the nyc stanza of package.json.
-  var config = require(path.resolve(this.cwd, './package.json')).config || {}
-  config = config.nyc || {}
+  var config
+  if (pkgPath) {
+    cwd = path.dirname(pkgPath)
+    var pkg = readPkg.sync(pkgPath, {normalize: false})
+    config = pkg.nyc || (pkg.config && pkg.config.nyc)
+  }
+  config = config || {}
+
+  this.cwd = cwd
```

Comment:
@jamestalmage once you've refactored the config loading logic into a helper, I say :shipit: